src/client.ts

```diff
@@ -33,50 +33,62 @@ class Client {
   }
 
   static async createChannel (): Promise<string> {
-    return await superagent.head('https://smee.io/new').redirects(0).catch((err) => {
+    return await request.head('https://smee.io/new').redirects(0).catch((err) => {
       return err.response.headers.location
     })
   }
 
   onmessage (msg: any): void {
     const data = JSON.parse(msg.data)
 
+    // Construct the new target URL merged query parameters.
     const target = new url.URL(this.target)
     const mergedQuery = Object.assign(target, data.query)
     target.search = querystring.stringify(mergedQuery)
-
     delete data.query
 
-    // Remove the host header, leaving it causes issues with SNI and TLS verification
+    // Remove the host header, leaving it causes issues with SNI and TLS
+    // verification.
     delete data.host
 
-    const req = superagent.post(url.format(target)).send(data.body)
-
+    // Remove the body from the data object so it won't get copied among all the
+    // other keys.
+    const body = data.body
     delete data.body
 
+    const req = request.post(url.format(target))
+    this.logger.info(`Initiating ${req.method} ${req.url}`)
+
     Object.keys(data).forEach(key => {
       void req.set(key, data[key])
     })
```

Comment:
Alternately,
```typescript
    Object.keys(data).forEach((key) => req.set(key, data[key]))
```

An analogous comment applies to lines 67 and 70 (although, it probably doesn't help line 67 much).

Reply:
Typescript doesn't seem to like this at all.

Reply:
Bummer.  My apologies.  (My grasp of Typescript's "features" doesn't extend much beyond my tenuous grasp of Javascript....)